### DISCUSSION TITLE
Identifying non-English audio files

### DISCUSSION THREAD

**Discussion by ThisIsTheOnlyUsernameAvailable** [openai/whisper#436] [Q&A] [answered]
Amazing project -- thank you for your hard work on it.

I have multiple directories that contain English, and non-English audio.  I'd like to be able to identify which files are non-English so that I can process them.

I see that Whisper will detect audio language if one is not specified at the command line, but I can't find a command option that will only perform this initial detection.  Does anyone know how if this is possible via either the command line, or Python?

Many thanks.

**Comment by jianfch** [comment, marked as answer]
```python
import whisper
model = whisper.load_model('base')
segment = whisper.pad_or_trim(whisper.log_mel_spectrogram('audio.mp3'), 3000).to(model.device)
probs = model.detect_language(segment)[1]
language = max(probs, key=probs.get)
print(language)
```
But to do just this on the command line, you'll have to modify the code first.

**Comment by ThisIsTheOnlyUsernameAvailable** [comment]
Thanks so much, @jianfch!
